src/main/java/com/shapesecurity/shift/parser/Parser.java

```diff
@@ -81,11 +85,27 @@ private void consumeSemicolon() throws JsError {
 
     @NotNull
     protected <T extends Node> T markLocation(@NotNull SourceLocation startLocation, @NotNull T node) {
-        // TODO: actually mark location
-//    node.loc = Maybe.just(new SourceSpan(Maybe.nothing(), startLocation, new SourceLocation(this.lastLine+1, this.lastIndex-this.lastLineStart, this.lastIndex)));
+        locations.put(
+                node,
```

Comment:
Oh, yes, I forgot the Java idiom of double-indenting continued lines. In that case, it is fine. But I would suggest pulling out some of the sub-expressions and assigning them to local variables for readability.